include/grpcpp/alts_context.h

```diff
@@ -0,0 +1,73 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPCPP_ALTS_CONTEXT_H
+#define GRPCPP_ALTS_CONTEXT_H
+
+#include <grpc/grpc_security_constants.h>
+#include <grpcpp/impl/codegen/security/auth_context.h>
+
+#include <memory>
+
+struct grpc_gcp_AltsContext;
+
+namespace grpc {
+
+typedef struct Versions {
```

Comment:
Sorry for missing this before. But you are defining something like `grpc::Versions` and `grpc::RpcProtocolVersions`, which are too broad.

One thing you can do is to do something like this (note I renamed places such as `Versions` to `Version`)

```
class AltsContext {
 public:
   struct RpcProtocolVersions {
     struct Version { int major_version; int minor_version; };
     Version max_rpc_version;
     Version min_rpc_version;
  };
  // Other stuff of the class
};
```